### PR TITLE
Bug fix - lagging in ``DataGroup`` html wrapper

### LINKED ISSUE #3041: A case of very slow `DataGroup` HTML repr
This takes several seconds, this is the first time I observe such as case:

image

### LINKED ISSUE #3045: `DataGroup` HTML raises exception for binned data with 2-D content buffer
```python
import scipp as sc

var = sc.ones(dims=['row', 'col'], shape=[100, 2])
binned = sc.bins(begin=sc.array(dims=['x'], values=[0, 30], unit=None), dim='row', data=var)
dg = sc.DataGroup(a=binned)
dg
```

### LINKED ISSUE #3059: DataGroup HTML repr raises with non-NumPy dtype
```python
import scipp as sc

sc.DataGroup(a=sc.array(dims=['row'], values=['a', 'b']))
```

### PR DESCRIPTION
**Pull request by YooSunYoung**
Fixes #3041 
Fixes #3059
Fixes #3045 

- Removed `dequeue` to retrieve the last element in inline preview

Here is the `timeit` result below.
The same wrapping that took 50.7 s now takes 586 ms with this update.

lagging_meas

### CODE REVIEW

File: src/scipp/html/formatting_datagroup_html.py [hunk 1]
```diff
@@ -22,46 +21,51 @@ def _format_shape(var: Union[Variable, DataArray, Dataset, DataGroup]) -> str:
...
+            return _format_atomic_value(var.values, maxidx=None)
+        elif var.ndim == 0:
+            return _format_atomic_value(var.value, maxidx=30)
+
+    if isinstance(var, Dataset):
+        view_iterable = list(var.items())
         format_item = _format_dictionary_item
         var_len = len(var)
+    elif isinstance(var, np.ndarray):
+        view_iterable = np.ravel(var)
```
Comment (SimonHeybrock) [inline src/scipp/html/formatting_datagroup_html.py:51]: We should not call `ravel`, it may cause a copy.

**Review by SimonHeybrock — CHANGES_REQUESTED** [review]

**Comment by SimonHeybrock** [comment]
@YooSunYoung There is a merge conflict. I will have another look once fixed.

**Comment by YooSunYoung** [comment]
Maybe I should add more cases in the unit test...

File: src/scipp/html/formatting_datagroup_html.py [hunk 2]
```diff
@@ -42,13 +42,9 @@ def _format_dictionary_item(name_item: tuple, maxidx: int = 10) -> str:
...
     """Inline preview of single or multi-dimensional data"""
-    if isinstance(var, Variable):
-        if var.ndim != var.values.ndim:
-            return _format_atomic_value(var.values, maxidx=None)
-        elif var.ndim == 0:
-            return _format_atomic_value(var.value, maxidx=30)
-
-    if isinstance(var, Dataset):
+    if isinstance(var, (Variable, DataArray)):
+        return inline_variable_repr(var).replace("<div>", '').replace("</div>", '')
```
Comment (SimonHeybrock) [inline src/scipp/html/formatting_datagroup_html.py:46]: What if there is a string value that contains this? Can you extract the part of `inline_variable_repr` before adding the HTML divs to a new function?

Comment (YooSunYoung) [reply src/scipp/html/formatting_datagroup_html.py:46]: That is a very good point... I will just slice the string instead of using `replace`

Comment (YooSunYoung) [reply src/scipp/html/formatting_datagroup_html.py]: I ended up updating the `formatting_html` to wrap the `inline_variable_repr` with `<div>` tag outside of the method itself.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**